### PR TITLE
Rebase thunderfish

### PR DESCRIPTION
**Pull request by jfsehuanes**
see #26

### CODE REVIEW

**Comment by fabiansinz** [comment]
I recommend not ignoring failing tests. Also, I'd really program more tests to make sure that the code is still valid after this much of restructuring.

**Comment by jgrewe** [comment]
tests fail due to missing IPython :)

**Comment by jfsehuanes** [comment]
We're working on the tests, Dr. Sinz. You have my word. :)

**Comment by fabiansinz** [comment]
Sounds good.

File: thunderfish/sortfishtype.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import numpy as np
+import peakdetection as pkd
```
Comment (janscience) [inline thunderfish/sortfishtype.py:2]: import standard libraries first, then derived stuff, and finally local modules:
numpy
pyplot
peakdetection

File: thunderfish/sortfishtype.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    ax[2].tick_params(axis='both', which='major', labelsize=fs - 2)
+    ax[2].set_title('Distribution of r-values', fontsize=fs + 2)
+
+    pass
+
+
+if __name__ == "__main__":
+    print("\nChecking sortfishtype module ...\n")
+    import sys
+    import matplotlib.pyplot as plt
```
Comment (janscience) [inline thunderfish/sortfishtype.py:140]: this is already imported

File: thunderfish/sortfishtype.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    if plot_data_func:
+        plot_data_func(data, samplerate, peak_idx, trough_idx, threshold, pulse_thres, suggestion, r_tr)
+
+    print('\nFish-type is %s. r-value = %.3f' % (suggestion, peak_ratio))
+    return suggestion, peak_ratio
+
+
+def plot_type_detection(data, rate, peak_idx, trough_idx, peakdet_th, r_value_th, type_suggestion, r_vals):
+    """ Plots the best-window, a zoomed index of it and the r-values that determine whether fish is pulse or wave-type.
```
Comment (janscience) [inline thunderfish/sortfishtype.py:68]: ok, I aggree, let's leave the plot function in the module.

Comment (janscience) [inline thunderfish/sortfishtype.py:68]: Ok, I accept the pull request. So that Till can proceed. But we will continue to work on the algorithms.

**Pull request merged**